### PR TITLE
Add return type enforcement to PSR-15

### PR DESCRIPTION
**Pull request by shadowhand**
Enforce return types, targeting PHP >= 7.0. Also renames files to reflect the new PSR title.

### CODE REVIEW

**Comment by shadowhand** [comment]
@php-fig/psr-15 as per discussion in Slack.

**Comment by shadowhand** [comment]
Should this have been merged without review from PSR-15 WG? Changes are more than just the PA settings.

**Comment by shadowhand** [comment]
Not a big deal though, everyone in the WG agreed to this change via Slack.

File: proposed/http-handlers/request-handlers.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ interface RequestHandlerInterface
      *
      * @return ResponseInterface
      */
-    public function handle(ServerRequestInterface $request);
+    public function handle(ServerRequestInterface $request): ResponseInterface
```
Comment (weierophinney) [inline proposed/http-handlers/request-handlers.md]: The trailing semicolon got dropped...

File: proposed/http-handlers/request-handlers.md [hunk 2]
```diff
@@ -125,9 +125,6 @@ interface MiddlewareInterface
      *
      * @return ResponseInterface
      */
-    public function process(
-        ServerRequestInterface $request,
-        RequestHandlerInterface $handler
-    );
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
```
Comment (weierophinney) [inline proposed/http-handlers/request-handlers.md]: Same here.

File: proposed/http-handlers/request-handlers.md (hunk 1 above)
Comment (shadowhand) [reply proposed/http-handlers/request-handlers.md]: Doh!

File: .pullapprove.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ groups:
                 - "HTTP"
             files:
                 - "proposed/http-factory/*"
-                - "proposed/http-middleware/*"
+                - "proposed/http-handlers/*"
```
Comment (harikt) [inline .pullapprove.yml]: When the directories are changed this breaks the link . Eg : https://groups.google.com/d/msg/php-fig/V12AAcT_SxE/8UTIIpzXAAAJ . 

Either we need to keep a README file in the place stating where it is moved or with necessary information.

**Pull request merged**